apps/home/routes.py

```diff
@@ -467,7 +467,8 @@ def view_labs(lab_id=None):
     
     labs = labs.all()
     running_labs = {lab.lab_id: lab.id for lab in LabInstances.query.filter_by(user_id=current_user.id, is_deleted=False).all()}
-    return render_template("pages/labs_view.html", labs=labs, lab_categories=lab_categories, running_labs=running_labs, segment="/labs/view")
+    completed_labs = {lab.lab_id for lab in LabInstances.query.filter_by(user_id=current_user.id, is_deleted=True).filter(LabInstances.finish_reason.isnot(None)).all()}
```

Comment:
Hi @IsraelPedreira, in order to optimize DB queries, can we leverage the previous query and search for all labs from the user (running or deleted) and then provide the correct semantic on the view component (Jinja template)?